### ISSUE TITLE
Input parameters with BlackBoxLLM

### ISSUE THREAD

**Issue by shere-khan** [zou-group/textgrad#103]
Hello,

FormattedLLMCall can take in inputs to replace parameters in a format string. But I don't see the same option for BlackBoxLLM. 

system_prompt_idea_generator = tg.Variable(
    "I want to be able to pass in parameters to this string {like_here} and set different values {like_here_2}",
    requires_grad=True,
    role_description="Structured system prompt to a language model that specifies the task."
)
model = tg.BlackboxLLM(LLM_IDEA_GENERATOR, system_prompt_idea_generator)

Say I have a model and I want to optimize a system prompt, and that system prompt takes in different named parameters depeding on which training data sample I'm using. How do i pass in these parameters to the format string then call the LLM?

Am I thinking about this incorrectly?

-Justin

**Comment by samgregson** [comment]
How is the system prompt dynamic? Is there a finite amount of prompts?  Couldn't you use some logic to route the llm call and have seperate variables that are optimised?

**Comment by mertyg** [comment]
You can use things like [FormattedLLMCall](https://github.com/zou-group/textgrad/blob/main/textgrad/autograd/llm_ops.py#L228)s where variables can be separately provided / optimized!

[Here's](https://github.com/zou-group/textgrad/blob/main/examples/notebooks/Tutorial-Test-Time-Loss-for-Code.ipynb) an example use case for a loss function:
